sympy/functions/special/bessel.py

```diff
@@ -677,64 +677,79 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
         _, e = arg.as_coeff_exponent(x)
 
         if e.is_positive:
-            term_one = ((-1)**(nu -1)*log(z/2)*besseli(nu, z))
-            term_two = (z/2)**(-nu)*factorial(nu - 1)/2 if (nu).is_positive else S.Zero
-            term_three = (-1)**nu*(z/2)**nu/(2*factorial(nu))*(digamma(nu + 1) - S.EulerGamma)
-            arg = Add(*[term_one, term_two, term_three]).as_leading_term(x, logx=logx)
-            return arg
+            if nu is S.Zero:
+                #Equation 9.6.8 of Abramowitz and Stegun (10th ed, 1972).
+                term = -log(z) - S.EulerGamma + log(2)
```

Comment:
@anutosh491 The reference I've quoted actually gives this without the constant term, but I've added the constant to avoid breaking an earlier test. It is unclear to me why Sympy says
```
In [15]: (1+log(x)).as_leading_term(x)
Out[15]: log(x) + 1
```

Reply:
Yes we do maintain the constants while calculating the leading terms for log based functions. Something like 
`log(2*x).as_leading_term(x, cdir=1)` can be further split into `(log(x) + log(2)).as_leading_term(x, cdir=1)` if that makes sense and we do need to take the log(2) term into account here for various reasons. Check out the examples/issues fixed here (#23592) if that further interests you but as of now I think what you did is correct, maintain the constant too.